tests/metrics/test_worker_integration.py

```diff
@@ -0,0 +1,239 @@
+                ),
+                call(
+                    "snitun.worker.peer_connections",
+                    1,
+                    {"protocol_version": "2"},
+                ),
+            ],
+        )
+
+
+def test_worker_with_noop_metrics() -> None:
+    """Test that worker works with no-op metrics (default)."""
+    worker = ServerWorker(
+        fernet_keys=["Wnng8SA8nnad6Q4YiZCFVMBMvxMJfn9pvMY7Wg_JBtw="],
+    )
+    assert worker._metrics_factory is not None
+    collector = worker._metrics_factory()
+    collector.gauge("test", 1.0)
+    assert not hasattr(collector, "_metrics")
+
+
+@pytest.mark.asyncio
+async def test_worker_metrics_task_lifecycle() -> None:
+    """Test that metrics task is properly managed during worker lifecycle."""
+    mock_metrics = MagicMock()
+    mock_factory = MagicMock(return_value=mock_metrics)
+    worker = ServerWorker(
+        fernet_keys=["Wnng8SA8nnad6Q4YiZCFVMBMvxMJfn9pvMY7Wg_JBtw="],
+        metrics_factory=mock_factory,
+        metrics_interval=0.1,
+    )
+
+    with patch.object(worker, "_loop", asyncio.get_event_loop()):
+        assert worker._metrics_task is None
+
+        mock_peers = MagicMock()
+        mock_peers.iter_peers = MagicMock(return_value=[])
+        worker._peers = mock_peers
+
+        await worker._async_init()
+
+        assert worker._metrics_task is not None
+        assert not worker._metrics_task.done()
+        assert worker._metrics is not None
+
+        await asyncio.sleep(0.15)
+
+        assert mock_metrics.gauge.called
+
+        worker._metrics_task.cancel()
+        with contextlib.suppress(asyncio.CancelledError):
+            await worker._metrics_task
+
+        assert worker._metrics_task.done()
+
+
+@pytest.mark.asyncio
+async def test_metrics_collection_without_peers() -> None:
+    """Test metrics collection when PeerManager is None."""
+    mock_metrics = MagicMock()
+    mock_factory = MagicMock(return_value=mock_metrics)
+
+    worker = ServerWorker(
+        fernet_keys=["Wnng8SA8nnad6Q4YiZCFVMBMvxMJfn9pvMY7Wg_JBtw="],
+        metrics_factory=mock_factory,
+    )
+    worker._metrics = mock_factory()
+    worker._peers = None
+
+    await worker._collect_and_report_metrics()
+
+    mock_metrics.gauge.assert_has_calls(
+        [
+            call("snitun.worker.peer_connections", 0),
+        ],
+    )
+
+
+@pytest.mark.asyncio
+async def test_metrics_collection_with_no_metrics_collector() -> None:
+    """Test that metrics collection handles None metrics collector gracefully."""
+    worker = ServerWorker(
+        fernet_keys=["Wnng8SA8nnad6Q4YiZCFVMBMvxMJfn9pvMY7Wg_JBtw="],
+    )
+
+    mock_peers = MagicMock()
+    mock_peers.iter_peers = MagicMock(return_value=[])
+    worker._peers = mock_peers
+    worker._metrics = None
+
+    try:
+        await worker._collect_and_report_metrics()
+    except BaseException:  # noqa: BLE001
+        pytest.fail("Metrics collection with None collector raised an error")
```

Comment:
```suggestion
   await worker._collect_and_report_metrics()
```
If it throws, let that bubble up, no need to catch and react on this.